### PR TITLE
Customize the docs theme with colors from Pylake logo

### PR DESCRIPTION
**Pull request by rpauszek**
**Why this PR?**
I thought it would be nice to have a slightly customized theme that reflects the colors from the logo and fixes some other eyesores. Help a colorblind guy out and make as many suggestions as you want, important thing is we know how to do it now

[docs build](https://lumicks-pylake.readthedocs.io/en/pylake_theme/)

Some before & afters:
g1742
image1586
image1600
image1614

### CODE REVIEW

File: docs/_static/custom.css [hunk 1]
```diff
@@ -1,7 +1,125 @@
...
+  background-color: var(--orange);
+}
+
+.rst-content .warning {
+  border-radius: 3px;
+  background-color: var(--lightorange);
+}
+
+.rst-content .warning a {
+  color: var(--orange);
```
Comment (tobiasjj) [inline docs/_static/custom.css:124]: I would make the text of links in a warning box darker, as I find the difference of the text of the link and the background to be uncomfortably small. I might have less rod cells ;-)
```suggestion
  color: var(--darkorange);
```

File: docs/_static/pylake_style.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        Name.Builtin: f'bold {colors["green"]}',
+        Name.Class: "underline",
+        Name.Namespace: f'bold {colors["blue"]}',
+        Name.Constant: colors["orange"],
+        Name.Decorator: f'bold {colors["orange"]}',
+        Name.Entity: f'bold {colors["pink"]}',
+        Name.Exception: f'bold {colors["pink"]}',
+        Name.Function: f'bold {colors["orange"]}',
+        Name.Tag: f'bold {colors["blue"]}',
+        Number: f'bold {colors["orange"]}',
```
Comment (tobiasjj) [inline docs/_static/pylake_style.py:67]: It feels uncomfortable reading numbers in code blocks. Probably also darken the orange a little bit? How do the others feel about that?

File: docs/_static/pylake_style.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        Operator: colors["blue"],
+        Operator.Word: f'bold {colors["blue"]}',
+        String: f' bold {colors["pink"]}',
+        String.Doc: "italic",
+        String.Escape: f'bold {colors["pink"]}',
+        String.Other: colors["orange"],
+        String.Symbol: f'bold {colors["pink"]}',
```
Comment (tobiasjj) [inline docs/_static/pylake_style.py:74]: Strings feel too bright (too pinkish :-) for my taste. It "shouts" when reading. Probably, again a _darker_ pink?

**Review by tobiasjj — CHANGES_REQUESTED** [review]
I like this PR, as we now easily can set custom color styles. I also like most of the modified colors, except for strings and numbers in code boxes and links in warning boxes. I would choose darker versions of the corresponding colors for these.

File: docs/_static/pylake_style.py (hunk 2 above)
Comment (aafkevandenberg) [reply docs/_static/pylake_style.py:67]: I agree, the numbers (and strings) now attract a lot of attention when reading code, while they are not the most important part of the code. Using darker colors would indeed help.

File: docs/_static/pylake_style.py (hunk 3 above)
Comment (aafkevandenberg) [reply docs/_static/pylake_style.py:74]: And was there an important reason for making strings bold? Using regular text would also make them stand out less.

**Review by aafkevandenberg — COMMENTED** [review]
Looks way better now!
Out of curiosity: did you really try to match the theme colors of LUMICKS? Did you just pick the colors that are visually closest to the theme colors?

File: docs/_static/custom.css (hunk 1 above)
Comment (JoepVanlier) [reply docs/_static/custom.css:124]: Yeah, I also find the links in those boxes hard to read.

File: docs/conf.py [hunk 4]
```diff
@@ -17,7 +17,9 @@
 import sphinx_rtd_theme
 
 sys.path.insert(0, os.path.abspath("./_ext"))
+sys.path.insert(0, os.path.abspath("./_static"))
 sys.path.insert(0, os.path.abspath(".."))
+# from ._static.pylake_style import PylakeDashStyle
```
Comment (JoepVanlier) [inline docs/conf.py:22]: Leftover comment?

File: docs/_static/custom.css [hunk 5]
```diff
@@ -67,7 +67,7 @@ code.xref.any.py {
   border-radius: 3px;
 }
 
-.rst-content dl[class]:not(.option-list):not(.field-list):not(.footnote):not(.glossary):not(.simple)>dt {
+html.writer-html5 .rst-content dl[class]:not(.option-list):not(.field-list):not(.footnote):not(.glossary):not(.simple)>dt {
```
Comment (JoepVanlier) [inline docs/_static/custom.css:70]: What is this added for?

**Review by JoepVanlier — CHANGES_REQUESTED** [review]
Looks nice. I agree regarding the orange thing.

Other than that, I noticed one minor leftover comment.

File: docs/_static/custom.css (hunk 5 above)
Comment (rpauszek) [reply docs/_static/custom.css:70]: not sure, but without it it doesn't seem to render properly. the css is pretty insane and impossible to make sense of ... I just used the browser inspect tool to figure out what things needed to be changed to get the desired effect

**Comment by rpauszek** [comment]
> Looks way better now! Out of curiosity: did you really try to match the theme colors of LUMICKS? Did you just pick the colors that are visually closest to the theme colors?

For the colors defined in the css I literally opened the pylake logo and used a color picker to get the exact hex codes, then used a shade & tint calculator to generate the lighter darker versions.

For the colors for syntax highlighting I messed with those a bit by hand to get something that looked better, and I agree the pink and orange need some more tweaking

**Review by tobiasjj — APPROVED** [review]
LGTM! 🚴

**Review by JoepVanlier — APPROVED** [review]
👍

**Pull request merged**